### PR TITLE
AST parser libs history

### PR DESCRIPTION
**Pull request by shayne-fletcher**
provide a little more color to the Background, Prior Art section.

### CODE REVIEW

File: proposals/accepted/000-ast-parser-libs.rst [hunk 1]
```diff
@@ -41,33 +41,69 @@ This section highlights various past and ongoing efforts accordingly.
...
+HLint_, the Haskell linter project developed continuously since 2008 is the most notable one, not just because its longstanding and wide use, but also because many of the developers that worked on the previous two libraries also worked on it --- use by HLint served as proof the libraries were fit-for-purpose.
+
+In June 2019,
+following the release of |ghc-lib-parser| and deprecation of |haskell-src_exts| earlier that year as described above,
+`HLint began the transitition <http://neilmitchell.blogspot.com/2019/06/hlints-path-to-ghc-parser.html>`_ to |ghc-lib-parser|.
+In May 2020, the release of HLint-3.0 which "uses the GHC parser" `was announced <http://neilmitchell.blogspot.com/2020/05/hlint-30.html>`_.
+
+Today, most users of |haskell-src-exts| have largely migrated from |haskell-src-exts| to |ghc-lib-parser| [#exampleghclibparserusers]_.
+But just because all these projects are using |ghc-lib-parser| doesn't mean everything is well.
+**Insert quote about maintainence overhead.**
```
Comment (Ericson2314) [inline proposals/accepted/000-ast-parser-libs.rst:98]: Thinking of what you put in the email to me about how much time the "ghc-lib-parser/ghc-lib/ghc-lib-parser-ex/hlint stack" was costing you, @shayne-fletcher.

File: proposals/accepted/000-ast-parser-libs.rst [hunk 2]
```diff
@@ -5,7 +5,7 @@ Split out AST and Parser libraries from GHC
 :Date: August 2023
 :Authors:
   John Ericson,
-  ???
+  Shayne Fletcher
```
Comment (Ericson2314) [inline proposals/accepted/000-ast-parser-libs.rst:8]: Can't forget that :)

**Review by Ericson2314 — APPROVED** [review]
I noticed that with your addition of the chronology of events, the ghc-lib-parser section heading no longer was where that library was introduced.

I thought about it, and it seemed like the only way to fix the topical separation of the sections while not messing up the chronology was to put HLint's story and its chronology in its own section. I then just went ahead and did, and added some text of why HLint has a "flagship" and frequent "guinea pig", is notable beyond just its status as a widely used project.

Let me know what you think @shayne-fletcher; feel free to disagree with that reorg, make some more changes, or just hit the merge button. :)

File: proposals/accepted/000-ast-parser-libs.rst (hunk 1 above)
Comment (shayne-fletcher) [reply proposals/accepted/000-ast-parser-libs.rst:98]: that's kind of you john. feel free to adapt "(ghc-lib-parser related maintenance) generally costs an hour or two a week and often more" in any way you see fit!

Comment (Ericson2314) [reply proposals/accepted/000-ast-parser-libs.rst:98]: It's not just kind, it is **crucial evidence** the modularity question isn't just an abstract one of code quality / library friendliness, but also a concrete one of reducing wasteful costs that you are stuck paying today!

**Pull request merged**